api.yml

```diff
@@ -185,66 +83,166 @@ paths:
           description: not supported by this implementation!
 components:
   schemas:
-    IssueRequest:
+    ConstructCredentialRequest:
       type: object
+      required:
+        - issuer
+        - vcTemplateReference
+        - claims
+        - output
       properties:
         issuer:
           type: string
-          description: Optional identifier for the issuer of the credential. If this is missing, it may be pre-configured or implied by other values in the request.
-          example: did:example:76e12ec712ebc6f1c221ebfeb1f
-        subject:
+          description: A reference to the issuer's pre-configured profile.
+        vcTemplateReference:
           type: string
-          description: Optional identifier for the subject of the credential. If this is missing, it may be pre-configured or implied by other values in the request, or there may not be an identifier for the subject of the credential.
-          example: did:example:ebfeb1f712ebc6f1c276e12ec21
-        types:
+          description: A reference to the pre-configured Verifiable Credential template.
+        claims:
           type: array
+          description: A set of claims about one or more subjects.
           items:
-            type: string
-          description: Optional value for the type(s) of a credential. If this is missing, it may be pre-configured or implied by other values in the request.
-          example: UniversityDegreeCredential
-        issuanceDate:
-          type: string
-          format: date-time
-          description: Optional value for the issuance date and time of the credential, to be used only when this value should not be automatically set to the current date and time.
-          example: '2010-01-01T19:23:24Z'
-        expirationDate:
+            $ref: '#/components/schemas/Claims'
+        output:
+          $ref: '#/components/schemas/Output'
+      example: {
+        "issuer": "did:example:123456677",
+        "vcTemplateReference": "https://issuer.oidp.uscis.gov/schemas/v1/prcard",
+        "claims": [{
+          "sub": "did:example:123456abcdefg",
+          "givenName": "JOHN",
+          "familyName": "SMITH",
+          "gender": "Male",
+          "image": "data:image/png;base64,iVBORw0KGgo...kJggg==",
+          "residentSince": "2015-01-01",
+          "lprCategory": "C09",
+          "lprNumber": "999-999-999",
+          "commuterClassification": "C1",
+          "birthCountry": "Bahamas",
+          "birthDate": "1958-07-17"
+        }],
+        "output": {
+          "encoding": "jsonld",
+          "encodingOptions": {
```

Comment:
These aren't really encoding options, or rather, not the same as the above. These are really options you're sending to the endpoint that have to do with the way the proof is constructed.

Reply:
Can change to "representationOptions" or anything else that might make more sense

Reply:
+1 :)

We'd need both `representationOptions` and `proofOptions` ... which then begs the question... why not just an "options" object... because we're going to be repeating xOptions, yOptions, and zOptions in the object. That was the original thinking behind putting all of this under `options`.